### ISSUE TITLE
[Bug]: Date precision

### ISSUE THREAD

**Issue by germanamz** [alpacahq/alpaca-trade-api-js#250]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

The implementation of the data stream casts timestamps with nanosecond precision to javascript `Date` object which has millisecond, this causes data loss.

### Expected Behavior

Have the original datetime string with nanosecond precision.

### SDK Version I encountered this issue in

alpaca-trade-api version: 3.0.1

### Steps To Reproduce

```markdown
1. Subscribe to any symbol using the data stream listener.
2. Get any trade event
```

### Filled out the Steps to Reproduce section?

- [X] I have entered valid steps to reproduce my issue or have attached a minimally reproducible case in code that shows my issue happening; and understand that without this my issue will be flagged as invalid and closed after 30 days.

**Comment by Azein** [comment]
@germanamz 
Good catch, I'm working on it for upcoming 4.0 For timestamps the way forward is pretty clear - additive change, `raw`-prefixed. Another interesting issue with the precision loss is probably crypto trade IDs, where I might need to introduce a breaking change.

**Comment by Azein** [comment]
Every market-data record — on both the live stream and the REST canonical shapes (Bar/Trade/Quote) — now carries an additive timestampRaw?: string: the original RFC-3339 timestamp with full nanosecond precision (e.g. "2024-01-02T03:04:05.678099211Z"), right alongside the convenient millisecond timestamp: Date.

While in there we also tackled the sibling 64-bit precision problem for ids: trade ids can exceed JavaScript's safe integer range (2^53), so trades now expose an exact idRaw?: string next to the numeric id — on the stream and the REST canonical trade accessors (getStockTrades/getCryptoTrades), so historical and live ids match. (Crypto trade ids in particular overflow a number.)

@germanamz If you are interested, you can check this out in alpha: https://www.npmjs.com/package/@alpacahq/alpaca-trade-api/v/4.0.0-alpha.3

**Comment by Azein** [comment]
4.0 is live, closing the issue.
